pkg/proxy/userspace/roundrobin.go

```diff
@@ -120,6 +120,14 @@ func isSessionAffinity(affinity *affinityPolicy) bool {
 	return true
 }
 
+// ServiceHasEndpoints checks whether a service entry has endpoints.
+func (lb *LoadBalancerRR) ServiceHasEndpoints(svcPort proxy.ServicePortName) bool {
+	lb.lock.Lock()
+	defer lb.lock.Unlock()
+	state, exists := lb.services[svcPort]
+	return exists && state != nil && len(state.endpoints) > 0
```

Comment:
some of the other code which uses `lb.services` explicitly deals with the `state == nil` check separately from the `exists` check (see `OnEndpointsUpdate`), while other code does not.  It looks like nothing currently ever assigns `nil` to an `lb.services` entry (although it could), so it's *probably* ok to remove the `state != nil` case.